command/agent/dns.go

```diff
@@ -547,17 +566,17 @@ RPC:
 	}
 
 	// If the network is not TCP, restrict the number of responses
-	if network != "tcp" && len(resp.Answer) > maxServiceResponses {
-		resp.Answer = resp.Answer[:maxServiceResponses]
+	if network != "tcp" {
+		wasTrimmed := trimAnswers(resp)
```

Comment:
@slackpad Apologies for bringing up an already merged PR, but it looks like this trimming is being done regardless of the enable_truncate setting. Shouldn't `d.config.EnableTruncate` be checked before `trimAnswers()` is called?